### DISCUSSION TITLE
Giscus integration in sphinx/readthedocs documentation

### DISCUSSION THREAD

**Discussion by 4flixt** [giscus/giscus#342] [Q&A] [answered]
Hi,

thank you for developing giscus and sharing it with us. I would love to integrate the feature in the documentation of our project [do-mpc](https://www.do-mpc.com/en/latest/).

This documentation is autogenerated with sphinx/readthedocs from our [Github repository](https://github.com/do-mpc/do-mpc).
It is slightly tricky to include custom html content in sphinx but it works. I tried to include the code snippet generated by Giscus now in the rst file [Discussion.rst](https://github.com/do-mpc/do-mpc/blob/develop/documentation/source/Discussion.rst):

```
**********
Discussion
**********

.. raw :: html

    <div class="giscus-container">
	<script src="https://giscus.app/client.js"
	        data-repo="do-mpc/do-mpc"
	        data-repo-id="MDEwOlJlcG9zaXRvcnkzNTk3NzI5MA=="
	        data-category="General"
	        data-category-id="MDE4OkRpc2N1c3Npb25DYXRlZ29yeTMyNzI5OTYz"
	        data-mapping="pathname"
	        data-reactions-enabled="1"
	        data-emit-metadata="0"
	        data-theme="light"
	        data-lang="en"
	        crossorigin="anonymous"
	        async>
	</script>
    </div>
```

This did build the following website on the develop version of the docs:

[https://www.do-mpc.com/en/develop/Discussion.html](https://www.do-mpc.com/en/develop/Discussion.html)

Unfortunately, I don't see any Giscus Plugin here. Inspecting the html document shows that the script was seemingly integrated as expected. 

Is it possible to make this work? Thank you so much in advance :)

Best,
4flixt

**Comment by 4flixt** [comment]
Note that I have currently tested:

```
data-mapping="number"
data-term="220"
```

pointing to this discussion: [https://github.com/do-mpc/do-mpc/discussions/220](https://github.com/do-mpc/do-mpc/discussions/220).

**Comment by laymonage** [comment]
Thanks for the report!

Apparently it seems because some of the variables used in the client script is already defined in the scope. I'll try to fix it soon.

![image](https://user-images.githubusercontent.com/6379424/149170130-637ecca0-cc7a-490f-8c86-4c492a8da256.png)

**Comment by 4flixt** [comment]
Hi @laymonage ! Thank you for the swift reply and for identifying the problem. I'll wait for further updates then :)

**Comment by laymonage** [comment, marked as answer]
Should be working now :)

![image](https://user-images.githubusercontent.com/6379424/149180008-199f3d89-4001-4a3f-b560-4821192ee845.png)

Thanks!

**Comment by 4flixt** [comment]
wow, you are fast! Thanks a lot :)
